modules/gapi/src/backends/streaming/gstreamingbackend.cpp

```diff
@@ -282,6 +284,26 @@ void GOCVBGR::Actor::extractRMat(const cv::MediaFrame& frame, cv::RMat& rmat)
             rmat = cv::make_rmat<cv::gimpl::RMatOnMat>(bgr);
             break;
         }
+        case cv::MediaFormat::GRAY:
+        {
+            std::call_once(m_warnFlagRGB,
```

Comment:
@AsyaPronina  As for me, i think that any approach in notification of user for potential drawbacks is a good idea.
But, IMHO, it SHOULD happens at graph construction phase using META information (if applicable). Despite on a comments an improvement of this part in a huge way seems exhaustive and not in scope of this PR :) 

P.S. This could be  easily handles using `static std::once ` in appropriate `case` section
P.P.S. In  system approach (to spread it widely for different components) it can be solved using Logger throttle technique (incapsulate counter in LOG)

Reply:
@sivanov-work , `static` shouldn't be the case here. We are creating `Actor` object for any usage of `cv::gapi::streaming::BGR` operation in code. We would warn once in case we use `cv::gapi::streaming::BGR` multiple times. What if we want to use 4 `cv::gapi:;streaming::BGR` operations for 4 input streams, two for `NV12`-s, one for  `GRAY`, one for `BGR`. There will be only 2 warnings about costly conversion from `NV12` and `GRAY`, but should be twice warned for `NV12` and once for `GRAY`. Do you agree?

Reply:
@dbudniko
>> Do you agree?
I do, it makes sense

>> finally agree
As i said before - if any other suggestions require huge work (more than adding one member in class) I would  **suppose to skip**  any other actions in that direction to do not affect scope of gray PR because it looks more important than warning discussions and should not be blocked due to other stuff refactoring too much.

But please feel yourself free to implement any other solution in order to improve warning mechanism: i have no objections